src/ast/semantic_analyser.cpp

```diff
@@ -1187,6 +1187,27 @@ void SemanticAnalyser::visit(Call &call)
     call.type = call.vargs->front()->type;
     call.type.SetAS(as);
   }
+  else if (call.func == "macaddr")
+  {
+    if (!check_nargs(call, 1))
+      return;
+
+    auto &arg = call.vargs->at(0);
+
+    if (!arg->type.IsIntTy() && !arg->type.IsArrayTy() &&
+        !arg->type.IsByteArray() && !arg->type.IsPtrTy() &&
+        !arg->type.IsRecordTy())
```

Comment:
`type::record` should only be structs not struct pointers so this should be removed right?

Reply:
Yep, good catch. Fixed in latest commit